internal/v1/handler_compose_image.go

```diff
@@ -705,12 +820,26 @@ func validateComposeRequest(cr *ComposeRequest) error {
 }
 
 func (h *Handlers) buildCustomizations(ctx echo.Context, cr *ComposeRequest, d *distribution.DistributionFile) (*composer.Customizations, error) {
+	res := &composer.Customizations{}
+
+	if cr.ImageRequests[0].ContentTemplate != nil {
+		templatePayloadRepositories, templateCustomRepositories, _, err := h.buildTemplateRepositories(ctx, *cr.ImageRequests[0].ContentTemplate)
+		if err != nil {
+			return nil, err
+		}
+		res.CustomRepositories = &templateCustomRepositories
+		res.PayloadRepositories = &templatePayloadRepositories
+	}
+
 	cust := cr.Customizations
 	if cust == nil {
+		// If no customizations requested, just return the repository snapshots resolved from the content template if one was specified
+		if res.CustomRepositories != nil && res.PayloadRepositories != nil {
+			return res, nil
+		}
 		return nil, nil
```

Comment:
I think this could be rewritten to make it a bit easier to read, how about:

```suggestion
	cust := cr.Customizations
	if cust == nil && cr.ImageRequests[0].ContentTemplate == nil {
		return nil, nil
	}

	res := &composer.Customizations{}
	if cr.ImageRequests[0].ContentTemplate != nil {
		templatePayloadRepositories, templateCustomRepositories, _, err := h.buildTemplateRepositories(ctx, *cr.ImageRequests[0].ContentTemplate)
		if err != nil {
			return nil, err
		}
		res.CustomRepositories = &templateCustomRepositories
		res.PayloadRepositories = &templatePayloadRepositories
	}
```

Reply:
hmmm i see a null pointer dereference when running TestComposeCustomizations with these changes, maybe there's something wrong with the cases i added here? do [these](https://github.com/osbuild/image-builder-crc/pull/1489/files#diff-8157e75e8106573de99e0c216371cf52ffe66f95cb956487c54ceafe6eaba7a8R2824) look right?

Reply:
ah, you're right, sorry misread this a bit. Updated the suggestion but willing to leave it as is. I think it's slightly easier to read if all the template logic is within the same code block.